### DISCUSSION TITLE
Error: Please set env DO_KEY_ID

### DISCUSSION THREAD

**Discussion by d3vx** [NativePHP/laravel#312] [Q&A]
Error: Please set env DO_KEY_ID (see https://www.electron.build/configuration/publish#spacesoptions)
    at SpacesPublisher.configureS3Options

**Comment by simonhamp** [comment]
I think we'll need a little more context to help you here

**Comment by mouradsme** [comment]
I think this has to do with the 'updater' option provided by NativePHP.
The default setting for NATIVEPHP_UPDATER_PROVIDER is 'spaces' which requires a key (DO_KEY_ID) from DigitalOcean (I suppose?).
You could use another provider (github or s3 for instance).

Github Example:
For github, just create a repository and push your project to that repo. 
Add the following to .env:
GITHUB_REPO='<repo_name>' # for example 'test-repo'
GITHUB_OWNER='<owner_username>'
GITHUB_TOKEN='github_pat_REST_OF_TOKEN....'
You can create the fine-grained token via Developer Settings > Personal Access Tokens
Make sure to set the right permissions for this token.

I hope this helps.

**Reply by mouradsme** [reply]
Note that upon a successful app publication (in github), if you go to your repo's release section, you should find a draft with assets (.exe files and .yml file) that you can release.
